package-res/ccc/core/base/slidingWindow.js

```diff
@@ -0,0 +1,233 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/*global axis_optionsDef:true*/
+  
+
+//CDF603
+
+def('pvc.visual.SlidingWindow', pvc.visual.OptionsBase.extend({
+
+    init: function(chart) {
+        this.base( chart, 'slidingWindow', 0, {byNaked: false} );
+    },
+
+    type: {
+        methods: {
+            props: ['interval','dimName','score', 'select']
+        }
+    },
+
+    methods: /** @lends pvc.visual.slidingWindow# */{
+        _initFromOptions: function() {
+            var o = this.option;
+            this.set({
+                interval :  o('Interval'),
+                dimName  :  o('DimName'),
+                score    :  o('Score'),
+                select   :  o('Select')
+            });
+        },
+
+        set: function(keyArgs) {
+
+            keyArgs = this._readArgs(keyArgs);
+
+            if(!keyArgs) {
+                if(this.interval != null && this.dimName != null && this.score != null  && this.select != null) return;
+            } else {
+                this.interval = pv.parseDatePrecision(keyArgs.interval, Number.MAX_VALUE);
+                this.dimName = keyArgs.dimName;
+                this.score = keyArgs.score ;
+                this.select = keyArgs.select ;
+            }
+
+        },
+
+        _readArgs: function(keyArgs) {
+            if(keyArgs) {
+                var out = {},
+                    any = 0,
+                    read = function(p) {
+                        var v = keyArgs[p];
+                        if(v != null)
+                            any = true;
+                        else
+                            v = this[p];
+
+                        out[p] = v;
+                    };
+
+                pvc.visual.SlidingWindow.props.forEach(read, this);
+
+                if(any) return out;
+            }
+        },
+
+        _defaultSlidingWindowScore: function(datum) { 
+            return datum.atoms[this.dimName].value; 
+        },
+
+        
+        _defaultSlidingWindowSelect: function(allData, remove) {
+                       
+            var data  = this.chart.data,
+                dName = this.dimName,
+                dim   = data.dimensions(dName),
+                mostRecent   = dim.max().value;
+
+            allData.forEach(function(datum) {
+                var datumScore = this.score(datum),
+                    result;
+
+                if(datumScore !== undefined && datumScore != null){
```

Comment:
`datumScore != null` would be enough. Testing for `v != null` is identical to testing for `v !== null && v !== undefined`. Plus, incorrect `) {` spacing.